### ISSUE TITLE
При обновлении сервиса "на горячую" HyperFastCgi либо падает, либо зависает и не выдаёт результат

### ISSUE THREAD

**Issue by altair86** [xplicit/HyperFastCgi#49]
Возможно есть правильная процедура обновления bin-файлов?

**Comment by xplicit** [comment]
В версии 0.4 только через  остановку процесса/обновление bin или web.config/запуск процесса, т. к. автоматический перезапуск после обновления до конца не реализован. В версии 0.3 работает автоматом.

**Comment by altair86** [comment]
Да, я установил 0.4. Кажется, у него всё в порядке. Или следует откатиться до 0.3?

Но общий вопрос: как следует правильно обновляться в случаях, когда нужно, чтобы сервер работает бесперебойно?

**Comment by xplicit** [comment]
Специфика ASP.NET такова, что при обновлении web.config или bin файлов происходит выгрузка работающего web-приложения из домена и веб-сервер (например IIS) загружает его снова, т. е. приложение в любом случае останавливает свое выполнение на какой-то момент, чтобы потом заработать. 
   Поэтому, если в HyperfastCgi запущено одно веб-приложение, то аналогичным действием для обновления будет:
- остановить hyperfastcgi
- обновить bin файлы и/или web.config
- запустить hyperfastcgi

Так же, как и в предыдущем случае, выполение будет остановлено на момент обновления. 

Можно ли сделать, чтобы http-запросы обрабатывались даже при перезапуске сервера во время обновления - да, можно, но ответ на вопрос как это сделать, довольно длинный и выходит за рамки issues hyperfastcgi.
